src/gtk/toplevel.cpp

```diff
@@ -1523,6 +1543,52 @@ wxTopLevelWindowGTK::DecorSize& wxTopLevelWindowGTK::GetCachedDecorSize()
     return size[index];
 }
 
+void wxTopLevelWindowGTK::GTKDoAfterShow()
+{
+    wxShowEvent showEvent(GetId(), true);
+    showEvent.SetEventObject(this);
+    HandleWindowEvent(showEvent);
+
+#ifdef __WXGTK3__
+    // Set the client size again if necessary, we should be able to do it
+    // correctly by now as the style cache should be up to date.
+    GTKUpdateClientSizeIfNecessary();
+#endif // __WXGTK3__
+}
+
+#ifdef __WXGTK3__
+
+void wxTopLevelWindowGTK::GTKUpdateClientSizeIfNecessary()
+{
+    if ( m_pendingFittingClientSizeFlags )
+    {
+        WXSetInitialFittingClientSize(m_pendingFittingClientSizeFlags);
+
+        m_pendingFittingClientSizeFlags = 0;
+    }
+}
+
+void wxTopLevelWindowGTK::SetMinSize(const wxSize& minSize)
+{
+    wxTopLevelWindowBase::SetMinSize(minSize);
+
+    // Explicitly set minimum size should override the pending size, if any.
+    m_pendingFittingClientSizeFlags &= ~wxSIZE_SET_MIN;
+}
+
+void wxTopLevelWindowGTK::WXSetInitialFittingClientSize(int flags)
+{
+    // In any case, update the size immediately.
+    wxTopLevelWindowBase::WXSetInitialFittingClientSize(flags);
+
+    // But if we're not shown yet, also redo it later if the decoration size
+    // may change.
+    if ( !IsShown() && m_updateDecorSize )
```

Comment:
Again, I don't think decoration size is involved, so the test for `m_updateDecorSize` seems wrong.